### PR TITLE
7643 - prevent noca on closed cases greater than 6 months

### PR DESCRIPTION
**Pull request by codyseibert**
covering the other flow where a petitioner can update their address and it shouldn't generate a notice of change address after a 6 month closed case

### CODE REVIEW

**Review by TomElliottFlexion — APPROVED** [review]

File: shared/src/business/useCases/updateContactInteractor.js [hunk 1]
```diff
@@ -95,7 +95,11 @@ exports.updateContactInteractor = async (
       oldData: oldCaseContact,
     });
 
-  if (!oldCaseContact.isAddressSealed && documentType) {
+  if (
+    !oldCaseContact.isAddressSealed &&
+    documentType &&
+    caseEntity.shouldGenerateNoticesForCase()
+  ) {
```
Comment (kkoskelin) [inline shared/src/business/useCases/updateContactInteractor.js:102]: Sorry to comment-after-merge.  Can we get test coverage on this file specifically for the new condition of `shouldGenerateNoticesForCase` ?

File: shared/src/business/entities/cases/Case.isCaseEligibleForService.test.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ const { Case } = require('./Case');
 const { CASE_STATUS_TYPES } = require('../EntityConstants');
 const { MOCK_CASE } = require('../../../test/mockCase');
 
-describe('isCaseEligibleForService', () => {
+describe('shouldGenerateNoticesForCase', () => {
```
Comment (kkoskelin) [inline shared/src/business/entities/cases/Case.isCaseEligibleForService.test.js:9]: Is it okay if we renamed this test file to match function name?

Comment (codyseibert) [reply shared/src/business/entities/cases/Case.isCaseEligibleForService.test.js:9]: will do

File: shared/src/business/useCases/updateContactInteractor.js (hunk 1 above)
Comment (codyseibert) [reply shared/src/business/useCases/updateContactInteractor.js:102]: will do

**Pull request merged**